CLONE_RESTORE.md

```diff
@@ -0,0 +1,189 @@
+
+To confirm if backups are being created by run `envdir /run/etc/wal-e.d/env wal-g backup-list` in the postgres pod created, you'll see a list of backups in s3
+
+<h2> Restoring backups from s3 using the clone method </h2>
+Although using an in-cluster database configuration that Harbor operator manages has its advantages, it also comes with a few disadvantages, a major one being some of the configurable
+parameters in the postgres CR are hard coded in the Harbor Operator, a few examples are the `patroni` and `clone`parameters (more on these later).
+
+We don't have direct access to the `clone` parameter in the postgres operator, e.g
+``` 
+clone:
+    uid: ""
+    cluster: ""
+```
+We have to find a way to simulate what this parameter does by passing in the needed parameters in the `pod-config` configmap. These are the parameters needed to trigger the clone scripts 
+on postgres pod startup
+
+
+PS: This is the ideal solution for the restoring a base backup as they were tested with regular postgres clusters.
+
+In other to restore backups from s3 these are the steps to follow:
+1. The password to the old cluster needs to be known and stored immediately it is created (This password is generated by the postgres operator)
+   `k get secret standby.<old-cluster-name>.credentials -n <namespace harbor CR was deployed in> -oyaml > secret.yaml`
+   The "creationTimestamp", "ownerReferences", "resourceVersion" and "uid" sections can be removed from the yaml file.
+2. Include the following parameters to the `pod-config` configmap used to create the initial cluster to tell the postgres operator to restore from an existing backup
+   ```
+   CLONE_SCOPE: "" # The name of the cluster you want to clone, ideally it is the folder name within the spilo folder in s3
+   CLONE_METHOD: "CLONE_WITH_WALE"
+   CLONE_HOST: "" # The same value as CLONE_SCOPE
+   CLONE_USER: "postgres"
+   CLONE_WAL_S3_BUCKET: "" # s3 bucket name the backup is stored in
+   CLONE_WAL_BUCKET_SCOPE_SUFFIX: "" # This is the id of the cluster, usually in the clone_scope folder in s3 (s3://<bucket-name>/spilo/<clone-scope>/<scope-suffix>) and it has to start with "/" e.g "/99e85b8a-3f92-4c86-8d42-5056348c1e7e"
+   CLONE_PORT: "5432"
+   CLONE_USE_WALG_RESTORE: "true"
+   USEWALG_RESTORE: "true"
+   CLONE_AWS_REGION: ""
+   CLONE_AWS_ACCESS_KEY_ID:
+   CLONE_AWS_SECRET_ACCESS_KEY:
+   ```
+3. You can use this reference manifest file to configure the harbor cluster and it's dependencies [here](https://docs.google.com/document/d/1g1sKP0fG-2qC9aLRfdMv75zJZ_B-S3I1B3upeBKgu0A)
+4. At this point, the pods will fail, reference problem 2 under the "Challenges faced while implementing this solution with Harbor Operator" to resolve it.
+5. Delete the `harbor-operator-postgres-operator-*` pod in the `harbor-operator-ns` namespace and wait for the pods to be recreated.
+6. Go to the configured harbor url (set as externalURL in the HarborCluster CR) to confirm your old data has been restored 
```

Comment:
Yes but a bit tricky.
- We can update the value of the CLONE parameters to that of the current running cluster so that it can do DR automatically on subsequent times the pods go down (This can be the default) the only thing to worry about is if the entire HarborCluser CR is deleted then we will need to update the `CLONE_WAL_BUCKET_SCOPE_SUFFIX` parameter to the old cluster ID and change it back to the current one when a successful backup has been done.